src/main/java/org/jabref/gui/preferences/appearance/AppearanceTab.java

```diff
@@ -27,6 +31,16 @@
 
     private final ControlsFxVisualizer validationVisualizer = new ControlsFxVisualizer();
 
+    // The fontSizeFormatter formats the input given to the fontSize spinner so that non valid values cannot be entered.
+    private TextFormatter<Integer> fontSizeFormatter = new TextFormatter<Integer>(new IntegerStringConverter(), 9,
+            c -> {
+                if (Pattern.matches("\\d*", c.getText())) {
+                    return c;
+                }
+                c.setText("0");
```

Comment:
I think returning `null` here is slightly better, at least according to the documentation "Returning null rejects the change." https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/TextFormatter.html#getFilter--

Reply:
See the comments in this thread. Returning null will produce an exception in the spinner